src/Morphir/Scala/Feature/Core.elm

```diff
@@ -929,15 +929,22 @@ curryConstructorArgs inScopeVars constructorType constructorFQName constructorAr
     in
     case ( constructorArgTypes, unspecifiedArgs ) of
         ( [], _ ) ->
-            Scala.Ref path (name |> Name.toTitleCase)
+            Scala.TypeAscripted
+                (Scala.Ref path (name |> Name.toTitleCase))
+                (mapType constructorType)
 
         ( [ _ ], [ _ ] ) ->
-            Scala.Ref path (name |> Name.toTitleCase)
+            Scala.TypeAscripted
+                (Scala.Ref path (name |> Name.toTitleCase))
+                (mapType constructorType)
 
         _ ->
             curryUnspecifiedArgs
                 unspecifiedArgs
-                (Scala.Ref path (name |> Name.toTitleCase))
+                (Scala.TypeAscripted
+                    (Scala.Ref path (name |> Name.toTitleCase))
+                    (mapType constructorType)
+                )
```

Comment:
I'm not sure if this will work as expected. It looks like we are adding the type ascription before we pass in the constructor arguments which won't be right. Adding some test cases with various number of constructor arguments will prove this.